### PR TITLE
Login via api

### PR DESCRIPTION
**Pull request by pechouc**
# What is done

This pull request essentially modifies the `manage_login_modal` callback in `main.py`, so as to check the credentials entered by the user through the login route of the API instead of the encrypted login correspondences file. As a consequence, all references to the `simple-crypt` library in the Dockerfile, the requirements, etc can be eliminated. I also added `requests` to the requirements. 

# What remains to be done

I could not find a way to retrieve the `group_id` of the logged-in user. I tried the `users/me` or simply `users` routes in a notebook, which work just fine, but I could not find any `group_id` field in the output. I added placeholders in the code to fetch the group id properly via the API (right now, it is set arbitrarily to 1) and I also added a small `group_correspondences` dictionary, that will give for each group around what point to center the map, as well as the zoom level to use. 

Maybe I was looking in the wrong direction for the `group_id` 🤔

### CODE REVIEW

File: app/main.py
Comment (Akilditu) [inline app/main.py:274]: It may sounds like a super dumb question, but when/where the fetch between the submitted username and pwd and this data object occurs ?

**Review by Akilditu — COMMENTED** [review]
Thanks for this PR, should we wait for the `group_id` answer from @frgfm before merge ?

File: app/main.py
Comment (chloeskt) [inline app/main.py:120]: We will need to hardcode this ? This dictionnary is never stored anywhere else ?

File: app/homepage.py
Comment (chloeskt) [inline app/homepage.py:361]: Should we keep this or remove it ?

File: app/main.py
Comment (pechouc) [reply app/main.py:274]: I am not sure to understand well the question but the username and password entered by the users are both a `State` of the callback, so they are passed as arguments to the `manage_login_modal` function. This allows to create the mini-dictionary `data` and to post it via `requests` on the next line. Is it what you wanted to know?

File: app/main.py
Comment (pechouc) [reply app/main.py:120]: If I understood well, what we will have for the tests in Ardèche on the API side is a way to fetch the `group_id` but there will not be fields like the center latitude and longitude and the zoom associated with it in the tables. So we would need to hardcode it, at least in a first version of the code (and as it will be very short, I guess it is simpler to hardcode it rather than storing it as a JSON file). But I might be wrong for the API and I am open to moving it to a static file if you prefer!

File: app/homepage.py
Comment (pechouc) [reply app/homepage.py:361]: Very nice catch, I commented this line for debugging but it should not stay as is indeed!

**Comment by pechouc** [comment]
> Thanks for this PR, should we wait for the `group_id` answer from @frgfm before merge ?

Yes, I guess so! 

And otherwise, I have pushed a new commit to correct for the backdrop issue noticed by @chloeskt and to add some comments in the new pieces of code.

File: app/main.py
Comment (Akilditu) [reply app/main.py:275]: Oh well did not see the State ....

File: app/main.py
Comment (pechouc) [reply app/main.py:120]: Will do in the afternoon!

File: app/main.py
Comment (frgfm) [reply app/main.py:298]: For now, in the same HTTP request: no
But I think on the API side, we're gonna add a "get my access" route : passing the token, it sends you back all non sensitive info (login, scope, group_id). Either way, consider that it will be done within the next days!

File: app/main.py
Comment (Akilditu) [inline app/main.py:105]: So this `dcc.Store` allows one API call once the app is instantiated and then stores it on a session scope right ?

**Review by Akilditu — APPROVED** [review]
Thanks a lot for the PR ! Ready to merge :)

File: app/main.py
Comment (pechouc) [reply app/main.py:105]: Yes, that would be the plan but I am not an expert of these storage components, so I hope it works!

**Pull request merged**
